java/src/main/java/ai/onnxruntime/OrtProviderOptions.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package ai.onnxruntime;
+
+import java.io.IOException;
+
+/** An abstract base class for execution provider options classes. */
+// Note this lives in ai.onnxruntime to allow subclasses to access the OnnxRuntime.ortApiHandle
+// package private field.
+public abstract class OrtProviderOptions implements AutoCloseable {
+  static {
+    try {
+      OnnxRuntime.init();
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to load onnx-runtime library", e);
+    }
+  }
+
+  protected final long nativeHandle;
```

Comment:
>nativeHandle [](http://example.com/codeflow?start=23&length=12)

I found that this handle needs to be accessible to other classes in the package (though not to the clients) so they can accept the instance of the class in its API and pass the native handle to ORT API.

Reply:
Yeah, protected in Java means it's this package + subclasses, so it is visible to everything in `ai.onnxruntime`.